### DISCUSSION TITLE
How does pnpm override the version of a sub-dependencies?

### DISCUSSION THREAD

**Discussion by ckvv** [pnpm/pnpm#4604] [Q&A] [answered]
I need to lock a sub-dependency version ！！！

**Comment by zkochan** [comment, marked as answer]
Using [overrides](https://pnpm.io/package_json#pnpmoverrides). For instance:

package.json:
```json
{
  "pnpm": {
    "overrides": {
      "foo": "^1.0.0",
      "quux": "npm:@myorg/quux@^1.0.0",
      "bar@^2.1.0": "3.0.0",
      "qar@1>zoo": "2"
    }
  }
}
```

**Reply by aolchik** [reply]
I'm trying the override a specific package with no success.
After setting overrides and installing both packages, I've got the following package.json

```
  "pnpm": {
    "overrides": {
      "netlify-cms": "@aolchik/netlify-cms@^2.10.192"
    }
  },
  "dependencies": {
    "@aolchik/netlify-cms": "^2.10.192",
    "netlify-cms": "link:@aolchik/netlify-cms@^2.10.192",
...
```

When running the application, I'm getting a ```Module not found``` error on netlify-cms.

Taking a look a node_modules, I have no folder installed/linked for netlify_cms.

Any idea of what I'm doing wrong?

**Reply by aolchik** [reply]
Problem apparently solved with the following config

```
  "pnpm": {
    "overrides": {
      "netlify-cms": "npm:@aolchik/netlify-cms@^2.10.192"
    }
  },
  "dependencies": {
    "@aolchik/netlify-cms": "^2.10.192",
    "netlify-cms": "^2.10.192",
...
  },
  "dependenciesMeta" : {
    "netlify-cms" : {
      "injected": true
    }
  }
```

**Reply by zkochan** [reply]
I don't think injecting netlify-cms does anything in this case.
